### PR TITLE
[ADD] 14.0_sms_no_automatic_delete

### PR DESCRIPTION
**Pull request by Kev-Roche**
Proposing a new module : sms_no_automatic_delete
It allows to save all sent sms and delete them after a specific delay.

### CODE REVIEW

**Comment by Kev-Roche** [comment]
cc @alexis-via, @sebastienbeau

File: sms_no_automatic_delete/models/sms_sms.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from odoo import api, models
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class SmsSms(models.Model):
+    _inherit = "sms.sms"
+
+    @api.model
+    def _purge(self, days):
+
```
Comment (sebastienbeau) [inline sms_no_automatic_delete/models/sms_sms.py]: remove this empty line

File: sms_no_automatic_delete/models/sms_sms.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        purge_day = datetime.now() - timedelta(days=days)
+        purge_day = purge_day.strftime(DEFAULT_SERVER_DATE_FORMAT)
+        sms = self.env["sms.sms"].search([("write_date", "<=", purge_day)])
```
Comment (sebastienbeau) [inline sms_no_automatic_delete/models/sms_sms.py]: You can directly pass a datetime here so 

```suggestion
        sms = self.env["sms.sms"].search([("write_date", "<=", datetime.now() - timedelta(days=days))])
```

File: sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        sms_91_days_ago = datetime.now() - timedelta(days=91)
+
+        sms_30_days_ago = sms_30_days_ago.strftime(DEFAULT_SERVER_DATE_FORMAT)
+        sms_61_days_ago = sms_61_days_ago.strftime(DEFAULT_SERVER_DATE_FORMAT)
+        sms_91_days_ago = sms_91_days_ago.strftime(DEFAULT_SERVER_DATE_FORMAT)
+
+        self.sms1 = self.sms_sms.create(
+            {
+                "state": "sent",
+                "write_date": sms_30_days_ago,
```
Comment (sebastienbeau) [inline sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py]: Same here pass a datetime directly ;)

```suggestion
                "write_date": datetime.now() - timedelta(days=30),
```

**Review by sebastienbeau — CHANGES_REQUESTED** [review]

File: sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py (hunk 3 above)
Comment (Kev-Roche) [reply sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py]: indeed, I did it in all files

**Review by sebastienbeau — APPROVED** [review]
LGTM code review

**Comment by Kev-Roche** [comment]
cc @hparfr

**Review by hparfr — APPROVED** [review]

**Pull request merged**